v2/arangodb/database_query.go

```diff
@@ -99,3 +102,63 @@ type QueryOptions struct {
 type QueryRequest struct {
 	Query string `json:"query"`
 }
+
+type ExplainQueryOptimizerOptions struct {
+	// A list of to-be-included or to-be-excluded optimizer rules can be put into this attribute,
+	// telling the optimizer to include or exclude specific rules.
+	//  To disable a rule, prefix its name with a "-", to enable a rule, prefix it with a "+".
+	// There is also a pseudo-rule "all", which matches all optimizer rules. "-all" disables all rules.
+	Rules []string `json:"rules,omitempty"`
+}
+
+type ExplainQueryOptions struct {
+	// If set to true, all possible execution plans will be returned.
+	// The default is false, meaning only the optimal plan will be returned.
+	AllPlans bool `json:"allPlans,omitempty"`
+
+	// An optional maximum number of plans that the optimizer is allowed to generate.
+	// Setting this attribute to a low value allows to put a cap on the amount of work the optimizer does.
+	MaxNumberOfPlans *int `json:"maxNumberOfPlans,omitempty"`
+
+	// Options related to the query optimizer.
+	Optimizer ExplainQueryOptimizerOptions `json:"optimizer,omitempty"`
+}
+
+type ExplainQueryResultExecutionNode map[string]interface{}
+type ExplainQueryResultExecutionCollection map[string]interface{}
+type ExplainQueryResultExecutionVariable map[string]interface{}
+
+type ExplainQueryResultPlan struct {
+	// Execution nodes of the plan.
+	Nodes []ExplainQueryResultExecutionNode `json:"nodes,omitempty"`
+	// List of rules the optimizer applied
+	Rules []string `json:"rules,omitempty"`
+	// List of collections used in the query
+	Collections []ExplainQueryResultExecutionCollection `json:"collections,omitempty"`
+	// List of variables used in the query (note: this may contain internal variables created by the optimizer)
+	Variables []ExplainQueryResultExecutionVariable `json:"variables,omitempty"`
+	// The total estimated cost for the plan. If there are multiple plans, the optimizer will choose the plan with the lowest total cost
+	EstimatedCost float64 `json:"estimatedCost,omitempty"`
+	// The estimated number of results.
+	EstimatedNrItems int `json:"estimatedNrItems,omitempty"`
+}
+
+type ExplainQueryResultExecutionStats struct {
+	RulesExecuted   int     `json:"rulesExecuted,omitempty"`
+	RulesSkipped    int     `json:"rulesSkipped,omitempty"`
+	PlansCreated    int     `json:"plansCreated,omitempty"`
+	PeakMemoryUsage uint64  `json:"peakMemoryUsage,omitempty"`
+	ExecutionTime   float64 `json:"executionTime,omitempty"`
+}
+
+type ExplainQueryResult struct {
+	Plan  ExplainQueryResultPlan   `json:"plan,omitempty"`
```

Comment:
Looks like `profile` field is missing https://www.arangodb.com/docs/3.10/http/aql-query.html#explain-and-parse-aql-queries

Reply:
'profile' is there only for `_api/cursor` response.